### ISSUE TITLE
Refactoring message when not all paths delegate to super() despite multiple bases each defining a method

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#7127]
### Current problem

#7007 adds these tests and alludes to a new warning we could emit to tell the difference:

```python
class Grandparent:
    def __repr__(self):
        return "I'm a grandparent"


class Parent1(Grandparent):
    def __repr__(self):
        return "I'm parent 1"


class Parent2(Grandparent):
    def __repr__(self):
        return "I'm parent 2"


class Child1(Parent1, Parent2):  # [hidden-parent-method]
    pass


class Child2(Parent1, Parent2):
    def __repr__(self):
        return "no warning, because reimplemented completely"


class Child3(Parent1, Parent2):
    def __repr__(self):
        print("could emit a warning because not all parent methods delegate to super") # <-- AT ISSUE
        return super().__repr__()
```

As long as we can tell it apart from:
```python
class HasClose:
    def close(self):
        """Example demonstrating better inheritance pattern"""


class Child6(HasClose):
    def close(self):
        print("Child6")
        super().close()


class Child7(HasClose):
    def close(self):
        print("Child7")
        super().close()


class Grandchild(Child6, Child7):
    def close(self):
        """All paths delegate to super -- no warning"""
        print("Grandchild")
        super().close()
```

### Desired solution

In the first example, `Child2.__repr__()` is hidden. In the second example, it's not, because all paths delegate to super.

### Additional context

See discussion at #6992 

/cc @NeilGirdhar @socketpair

**Comment by NeilGirdhar** [comment]
Thank you for creating this.  In an ideal world, there would be a decorator in `typing` to mark a base class method as "all implementations delegate to super".  This would apply to methods like `__init__`, `__enter__`, `__exit__`, `numpy.ndarray.__array_finalize__`, `HasClose.close`, etc.
